app/controllers/application_controller.rb

```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
   include ApplicationHelper
+
+  helper_method :current_user
+
+  def current_user
+    if session[:user_id]
+      @current_user ||= User.find(session[:user_id])
+    else
+      @current_user = nil
+    end
```

Comment:
```suggestion
    @current_user ||= if session[:user_id]
      User.find(session[:user_id])
    end
```

i think this will work